src/Grandeljay/Availability/Config.php

```diff
@@ -54,20 +56,122 @@ private function loadConfig(): void
         die('Missing config.json. Please refer to README.md.' . PHP_EOL);
     }
 
+    /**
+     * Processes the passed raw config and returns it in normalised form.
+     *
+     * Note: This function doesn't do any validation.
+     *
+     * @param array $config The raw config to normalise. This is essentially
+     *                      just the decoded json string.
+     *
+     * @return array The normalised config.
+     */
+    private function normaliseConfig(array $rawConfig): array
+    {
+        $normalisedConfig = $rawConfig; // Create a copy.
+
+        $normalisedConfig['directoryAvailabilities']  = $this->normaliseAvailabilitiesDir($rawConfig['directoryAvailabilities']);
+        $normalisedConfig['maxAvailabilitiesPerUser'] = $rawConfig['maxAvailabilitiesPerUser'] ?? 100;
+        $normalisedConfig['defaultDay']               = $rawConfig['defaultDay'] ?? "monday";
+        $normalisedConfig['defaultTime']              = $rawConfig['defaultTime'] ?? "19:00";
+        $normalisedConfig['eventName']                = $rawConfig['eventName'] ?? "Dota 2";
+        $normalisedConfig['logLevel']                 = $rawConfig['logLevel'] ?? "Info";
+
+        return $normalisedConfig;
+    }
+
+    private function normaliseAvailabilitiesDir(?string $inputDir): string
+    {
+        $default = '$HOME/.local/share/discord-availability/availabilities';
+
+        $dir = $inputDir ?? $default;
+        $dir = $this->getPathWithEnvironmentVariable($dir);
+        $dir = $this->normalisePath($dir);
+
+        return $dir;
+    }
+
+    /**
+     * Returns an absolute path.
+     *
+     * Unlike `realpath` this function also works on paths that don't point to
+     * an existing file.
+     *
+     * Also, symlinks are not resolved.
+     *
+     * @param string $path
+     *
+     * @return string
+     */
+    private function normalisePath(string $path): string
+    {
+        if ($this->isPathAbsolute($path)) {
+            return $path;
+        }
+
+        $cwd = getcwd();
+
+        if (!$cwd) {
+            die('Could not determine current working directory.');
+        }
+
+        $segments = array($cwd, $path);
+        $absolutePath = implode(DIRECTORY_SEPARATOR, $segments);
```

Comment:
OK, this is the last nit nitpick then I'm ready to merge. Can you please align these equal sings too? Thanks for everything, looks good!

Reply:
Ok, done.